package-res/resources/web/prompting/api/UiAPI.js

```diff
@@ -15,8 +15,25 @@
  *
  */
 
+/**
+ * The Prompting UI API Class
+ * Provides general functions for working with UI of prompting.
+ *
+ * @name UiAPI
+ * @class
+ */
 define([], function() {
   return function(api) {
-
+    /**
+     * Makes visible the progress indicator by calling function Dashboard#showProgressIndicator.
+     *
+     * @name UiAPI#showProgressIndicator
+     * @method
+     * @example
+     *     api.ui.showProgressIndicator();
+     */
+    this.showProgressIndicator = function() {
+      api.operation._getPromptPanel().dashboard.showProgressIndicator();
```

Comment:
I'd probably avoid dashboard references to creep up to this level of the API.
This code assumes that a prompt panel will always contain a dashboard object that implements a showProgressIndicator method.

If we are targeting being able to change the prompt renderer from CDF to something else, that would mean that we'd have to change this function to achieve that, which feels wrong.

The easy fix would be for promptPanel to encapsulate how it shows/hides the progress indicator, therefore making the dashboard reference internal to it.